Added Colab TPU support with Colab Notebook and modified repo

aditya-malte opened the pull request:
I have made a colab notebook to allow for easy use of Google's Colab TPU. 
The same has been tried and test successfully for Colab TPU. 

(This includes additional things such as downloading weights to gcs buckets but keeping spiece file locally) 

Thank you

aditya-malte committed:
Update run_classifier for Colab Notebook

aditya-malte committed:
Update README.md

aditya-malte committed:
Update README.md

aditya-malte committed:
Created using Colaboratory

aditya-malte committed:
Delete End_To_End_Finetuning_of_XLNet_Using_TPU.ipynb

aditya-malte committed:
Created using Colaboratory

aditya-malte committed:
Update README.md

aditya-malte committed:
Update README.md

aditya-malte committed:
Update README.md

aditya-malte committed:
Update README.md

aditya-malte committed:
Delete End_To_End_Finetuning_of_XLNet_Using_TPU.ipynb

aditya-malte committed:
Created using Colaboratory

aditya-malte committed:
Created using Colaboratory

aditya-malte committed:
Update Colab_XLNet_FineTuning.ipynb

aditya-malte committed:
Update Colab_XLNet_FineTuning.ipynb

aditya-malte committed:
Solved MODEL_DIR, OUTPUT_DIR glitch

aditya-malte committed:
final working version

aditya-malte commented:
Hello,
I have uploaded the final working version after making MODEL/OUTPUT_DIR separate. Please check
Thank you

kimiyoung commented:
@aditya-malte Thanks for your contribution. It would be nice if you could do the following:
- merge your changes with the original `configure_tpu` function to support all the cases;
- remove other unnecessary changes such as README and other code changes in `run_classifier.py`;
- I noticed you add a trac dataset but without a processor. Could you change that to IMDB or STS-B so that it's consistent with the main examples in README.
- move the contents in README to the notebook.

aditya-malte commented:
Hello @kimiyoung,
Yes, I'll make the changes shortly and update you on it

aditya-malte committed:
Update README.md

aditya-malte committed:
Update run_classifier.py

aditya-malte committed:
Update README.md

aditya-malte committed:
Update run_classifier.py

aditya-malte committed:
Update run_classifier.py

aditya-malte committed:
Update model_utils.py

aditya-malte committed:
Created using Colaboratory

aditya-malte committed:
Update run_classifier.py

aditya-malte committed:
Created using Colaboratory

aditya-malte committed:
Created using Colaboratory

aditya-malte committed:
Created using Colaboratory

aditya-malte commented:
Hello @kimiyoung ,
I have made the requisite changes the you mentioned and also added IMDB as an example. Its running successfully, please check.
Thank you

aditya-malte committed:
Created using Colaboratory

aditya-malte committed:
Created using Colaboratory

aditya-malte commented:
Working perfectly for IMDB dataset for max_seq =128 and batch_size 64. Currently testing how far I can push the Colab TPU by increasing max_seq and/or batch_size

aditya-malte committed:
Changed "my_bert_5" to "YOUR_BUCKET_NAME"

aditya-malte committed:
Created using Colaboratory

aditya-malte committed:
Update Colab_XLNet_FineTuning.ipynb

aditya-malte committed:
Created using Colaboratory

aditya-malte committed:
Created using Colaboratory

aditya-malte committed:
Created using Colaboratory

aditya-malte commented:
Gives near SOTA results for eval_accuracy(IMDB) result equal to 0.9512. with batch size 32 and max_seq=256(Increasing max_seq further throws OOM error). Training and evaluation combined get completed within an hour or so.
Thank you

aditya-malte committed:
Created using Colaboratory

aditya-malte committed:
Minor Update

aditya-malte committed:
Merge pull request #1 from zihangdai/master

Sync with source repo

aditya-malte committed:
Created using Colaboratory

aditya-malte committed:
Rename Colab_XLNet_FineTuning.ipynb to colab_imdb_tpu

aditya-malte committed:
Delete Colab_XLNet_FineTuning.ipynb

aditya-malte committed:
Rename colab_imdb_tpu to colab_imdb_tpu.ipynb

aditya-malte commented:
Hello @kimiyoung ,
I have synchronized the fork to reflect the changes that have happened to the source repository.
Any updates from your side?
Thank you

Best Regards,
Aditya Malte

aditya-malte committed:
Update run_classifier.py

aditya-malte committed:
Update run_classifier.py

aditya-malte committed:
Update run_classifier.py

aditya-malte committed:
Update README.md

aditya-malte committed:
Fixed colab link

aditya-malte committed:
Update

aditya-malte committed:
Update README.md

aditya-malte committed:
Sync readme with original

aditya-malte commented:
Hello @kimiyoung ,
I have updated my repo again to reflect changes with your latest xlnet repo,
To summarise:
1) run_classifier remains exactly the same except the added flags: use_colab(default=False) and tpu_address(default=None)
2) modified model_utils make corresponding changes if use_colab is set ( a simple if condition), rest remains exactly the same.
3) README is identical, except for two lines saying that colab TPU version also available
4)Colab TPU Notebook in notebooks folder.

Thanks

aditya-malte committed:
Merge pull request #2 from zihangdai/master

Sync(update) with source repo changes

aditya-malte commented:
The above merge PR, is in order to update my repo with all the changes made in yours

cedspam committed:
Update train.py

aditya-malte committed:
Merge pull request #6 from cedspam/patch-1

Update train.py
